javatests/arcs/core/storage/DirectStoreMuxerTest.kt

```diff
@@ -10,19 +11,70 @@ import arcs.core.data.SchemaFields
 import arcs.core.data.util.toReferencable
 import arcs.core.storage.api.DriverAndKeyConfigurator
 import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.testutil.MockDriver
+import arcs.core.storage.testutil.MockDriverProvider
 import arcs.core.storage.testutil.testWriteBackProvider
 import com.google.common.truth.Truth.assertThat
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.coroutineScope
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.After
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.junit.runners.JUnit4
+import kotlin.test.assertFailsWith
 
 @RunWith(JUnit4::class)
 class DirectStoreMuxerTest {
 
+  private val storageKey: RamDiskStorageKey = RamDiskStorageKey("test")
+  private val schema = Schema(
+    emptySet(),
+    SchemaFields(
+      singletons = mapOf(
+        "name" to FieldType.Text,
+        "age" to FieldType.Int
+      ),
+      collections = emptyMap()
+    ),
+    "abc"
+  )
+
+  private fun createDirectStoreMuxer(coroutineScope: CoroutineScope): DirectStoreMuxer<CrdtEntity.Data, CrdtEntity.Operation, CrdtEntity> {
+    return DirectStoreMuxer<CrdtEntity.Data, CrdtEntity.Operation, CrdtEntity>(
+      storageKey,
+      backingType = EntityType(schema),
+      coroutineScope = coroutineScope,
+      writeBackProvider = ::testWriteBackProvider,
+      devTools = null
+    )
+  }
+
+  private fun DirectStoreMuxer<CrdtEntity.Data, CrdtEntity.Operation, CrdtEntity>.getEntityDriver(
+    id: ReferenceId
+  ): MockDriver<CrdtEntity.Data> =
+    requireNotNull(stores[id]).store.driver as MockDriver<CrdtEntity.Data>
+
+  private fun createPersonEntityCrdt(name: String, age: Int): CrdtEntity = CrdtEntity(
+    CrdtEntity.Data(
+      versionMap = VersionMap("me" to 1),
+      singletons = mapOf(
+        "name" to CrdtSingleton(
+          initialVersion = VersionMap("me" to 1),
+          initialData = CrdtEntity.Reference.buildReference(name.toReferencable())
+        ),
+        "age" to CrdtSingleton(
+          initialVersion = VersionMap("me" to 1),
+          initialData = CrdtEntity.Reference.buildReference(age.toReferencable())
+        )
+      )
+    )
+  )
```

Comment:
Helper functions like these for tests should typically go at the bottom of the test class... (making the tests themselves more front-and-center for the reader)